draft-ietf-mls-protocol.md

```diff
@@ -566,107 +654,67 @@ above:
+* Zero or more encrypted copies of the path secret corresponding to
+  the node
 
-The secret value is encrypted for the subtree corresponding to the
+The path secret value for a given node is encrypted for the subtree corresponding to the
 parent's non-updated child, i.e., the child on the copath of the leaf node.
-There is one encrypted secret for each public key in the resolution
+There is one encrypted path secret for each public key in the resolution
 of the non-updated child.  In particular, for the leaf node, there
 are no encrypted secrets, since a leaf node has no children.
 
 The recipient of an update processes it with the following steps:
 
-1. Compute the updated secret values
+1. Compute the updated path secrets
   * Identify a node in the direct path for which the local member
     is in the subtree of the non-updated child
   * Identify a node in the resolution of the copath node for
     which this node has a private key
-  * Decrypt the secret value for the parent of the copath node using
+  * Decrypt the path secret for the parent of the copath node using
     the private key from the resolution node
-  * Derive secret values for ancestors of that node using the KDF keyed with the
-    decrypted secret
-2. Merge the updated secrets into the tree
+  * Derive path secrets for ancestors of that node using the
+    algorithm described above
+2. Merge the updated path secrets into the tree
   * Replace the public keys for nodes on the direct path with the
     received public keys
-  * For nodes where an updated secret was computed in step 1,
-    replace the secret value for the node with the updated value
+  * For nodes where an updated path secret was computed in step 1,
+    compute the corresponding node secret and node key pair and
+    replace the values stored at the node with the computed values.
 
-For example, suppose we had the following tree:
-
-~~~~~
-      G
-    /   \
-   /     \
-  E       _
- / \     / \
-A   B   C   D
-~~~~~
+For example, in order to communicate the example update described in
+the previous section, the sender would transmit the following
+values:
 
-If an update is made along the direct path B-E-G, then the following
-values will be transmitted (using pk(X) to represent the public key
-corresponding to the secret value X and E(K, S) to represent
-public-key encryption to the public key K of the secret value S):
+| Public Key | Ciphertext(s)                    |
+|:-----------|:---------------------------------|
+| pk(ns[2])  | E(pk(C), ps[2]), E(pk(D), ps[2]) |
+| pk(ns[1])  | E(pk(A), ps[1])                  |
+| pk(ns[0])  |                                  |
 
-| Public Key | Ciphertext(s)            |
-|:-----------|:-------------------------|
-| pk(G)      | E(pk(C), G), E(pk(D), G) |
-| pk(E)      | E(pk(A), E)              |
-| pk(B)      |                          |
+Here pk(X) represents the public key corresponding derived from the
```

Comment:
I had trouble to understand the sentence for some reason... :)

Reply:
Probably because it's poorly written ;)  I rewrote it to hopefully be clearer.